src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -63,24 +63,18 @@ internal static string EnsureTrailingDirectorySeparator(string fullPath)
         //
         public static bool Exists(string path)
         {
-            try
+            if (string.IsNullOrEmpty(path) || path.Contains('\0'))
+                return false;
+            
+            if (!path.StartsWith(@"\\?\") && !path.StartsWith(@"\??\"))
```

Comment:
Isn't all of this Windows-specific?

Reply:
Yeah it is just windows specific

Reply:
Ok, thanks, then it shouldn't be in this shared portion of the implementation.